internal/server/boltdbstate/task.go

```diff
@@ -274,6 +308,67 @@ func (s *State) taskDelete(
 	return nil
 }
 
+/*
+func (s *State) taskCancel(
```

Comment:
do we have plans for this code or was it left in by accident?

Reply:
I should leave a comment here - but I left this purposefully commented out due to my other note above `TaskCancel`. In the future if we ever fix `TaskCancel` to cancel these jobs within a single transaction, we can just uncomment this and get it most of the way there.